### PR TITLE
Remove user-owned repositories from GraphQL and client side

### PR DESCRIPTION
**Pull request by mrnugget**
This removes from the GraphQL API and from client-side code:

* user-added repositories
* post-sign-up flow

I got the go-ahead on both of these removals from @ryphil. 

## Test plan

- Existing tests
- Manual sanity check by running app locally and clicking around user settings
- main dry run: https://buildkite.com/sourcegraph/sourcegraph/builds/171295

### CODE REVIEW

File: client/web/src/auth/ExternalsAuth.tsx [hunk 1]
```diff
@@ -73,7 +71,7 @@ const ExternalsAuth: React.FunctionComponent<React.PropsWithChildren<ExternalsAu
         <>
             {githubProvider && (
                 <Link
-                    to={maybeAddPostSignUpRedirect(githubProvider.authenticationURL)}
+                    to={githubProvider.authenticationURL || ''}
```
Comment (eseliger) [inline client/web/src/auth/ExternalsAuth.tsx:74]: I don't think a link that might have an empty to should exist, if it really can be empty, consider using LinkOrSpan instead

File: client/web/src/auth/ExternalsAuth.tsx [hunk 2]
```diff
@@ -88,7 +86,7 @@ const ExternalsAuth: React.FunctionComponent<React.PropsWithChildren<ExternalsAu
 
             {gitlabProvider && (
                 <Link
-                    to={maybeAddPostSignUpRedirect(gitlabProvider.authenticationURL)}
+                    to={gitlabProvider.authenticationURL || ''}
```
Comment (eseliger) [inline client/web/src/auth/ExternalsAuth.tsx:89]: same as above

File: client/web/src/auth/SignUpForm.tsx [hunk 3]
```diff
@@ -276,7 +276,7 @@ export const SignUpForm: React.FunctionComponent<React.PropsWithChildren<SignUpF
                             // here because this list will not be updated during this component's lifetime.
                             <div className="mb-2" key={index}>
                                 <Button
-                                    to={maybeAddPostSignUpRedirect(provider.authenticationURL)}
+                                    to={provider.authenticationURL || ''}
```
Comment (eseliger) [inline client/web/src/auth/SignUpForm.tsx:279]: same as above

**Review by eseliger — APPROVED** [review]
🔥

File: client/web/src/user/settings/sidebaritems.ts [hunk 4]
```diff
@@ -36,16 +36,6 @@ export const userSettingsSideBarItems: UserSettingsSidebarItems = [
         exact: true,
         condition: showAccountSecurityPage,
     },
-    {
-        label: 'Code host connections',
-        to: '/code-hosts',
-        condition: userExternalServicesEnabled,
-    },
-    {
```
[the remark this answers is not in the document]
Comment (mrnugget) [reply client/web/src/user/settings/sidebaritems.ts:44]: Do you mean this? https://github.com/sourcegraph/sourcegraph/pull/41350/files/387a9b0ff1f26c0c106165074f91d30526885506#diff-efea4ba057c39e1c2a64cec66858dab0e93bbe8974d6f7e3c45a81782f172be6L116-L123

[the remark this answers is not in the document]
Comment (eseliger) [reply client/web/src/user/settings/sidebaritems.ts:44]: ah forgot to remove the comment after I saw this, sorry!

File: client/web/src/auth/ExternalsAuth.tsx (hunk 1 above)
Comment (eseliger) [reply client/web/src/auth/ExternalsAuth.tsx:74]: Hmm I think honestly it was wrong. It's an anti-pattern to have links without an href

Comment (mrnugget) [reply client/web/src/auth/ExternalsAuth.tsx:74]: I somehow assumed it would be a link to `'/'`?

Comment (eseliger) [reply client/web/src/auth/ExternalsAuth.tsx:74]: Oh, maybe?

**Review by unknwon — APPROVED** [review]
Go files LGTM ...

File: client/web/src/auth/ExternalsAuth.tsx (hunk 1 above)
Comment (unknwon) [reply client/web/src/auth/ExternalsAuth.tsx:74]: Not sure how it was coded, but pretty sure `githubProvider.authenticationURL` won't be empty.

Comment (mrnugget) [reply client/web/src/auth/ExternalsAuth.tsx:74]: Yeah, looks like that's the case: https://github.com/sourcegraph/sourcegraph/blob/6f9b3d7f853de287ece39d1ab41555d5a7a1fdb4/cmd/frontend/internal/app/jscontext/jscontext.go#L156

I'm going to change it from `authenticationURL?: string` to `authenticationURL: string`

**Review by pjlast — APPROVED** [review]
huge

File: client/web/src/auth/ExternalsAuth.tsx [hunk 5]
```diff
@@ -63,17 +61,17 @@ const ExternalsAuth: React.FunctionComponent<React.PropsWithChildren<ExternalsAu
     // for other auth providers such SAML, OpenID, Okta, Azure AD, etc.
 
     const githubProvider = context.authProviders.find(provider =>
-        provider.authenticationURL?.startsWith('/.auth/github/login?pc=https%3A%2F%2Fgithub.com%2F')
+        provider.authenticationURL.startsWith('/.auth/github/login?pc=https%3A%2F%2Fgithub.com%2F')
```
Comment (kopancek) [inline client/web/src/auth/ExternalsAuth.tsx:64]: Was this change intentional? If yes, why is it needed? IMO the code works just fine even with the `?` operator.

File: client/web/src/auth/ExternalsAuth.tsx [hunk 6]
```diff
@@ -63,17 +61,17 @@ const ExternalsAuth: React.FunctionComponent<React.PropsWithChildren<ExternalsAu
...
     )
     const gitlabProvider = context.authProviders.find(provider =>
-        provider.authenticationURL?.startsWith('/.auth/gitlab/login?pc=https%3A%2F%2Fgitlab.com%2F')
+        provider.authenticationURL.startsWith('/.auth/gitlab/login?pc=https%3A%2F%2Fgitlab.com%2F')
```
Comment (kopancek) [inline client/web/src/auth/ExternalsAuth.tsx:67]: Same as above

**Review by kopancek — APPROVED** [review]

File: client/web/src/auth/ExternalsAuth.tsx (hunk 5 above)
Comment (mrnugget) [reply client/web/src/auth/ExternalsAuth.tsx:64]: See the discussion above: https://github.com/sourcegraph/sourcegraph/pull/41350#discussion_r963827695

TL;DR: I changed `authenticationURL` from `string | undefined` to `string` because it's never undefined. After that change the `?` resulted in a warning by ESLint.
